### PR TITLE
Extend README

### PR DESCRIPTION
**Pull request by tdsmith**
Fixes https://github.com/mozilla/jetstream/issues/39.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,9 +4,152 @@ Custom configs for experiments analyzed in [jetstream](https://github.com/mozill
...
+Configuration files have three main sections:
+`[experiment]`, `[metrics]`, and `[data_sources]`.
+
+Examples of every value you can specify in each section are given below.
+You do not need to specify everything!
+Jetstream will take values from Experimenter
+and combine them with a reasonable set of defaults.
+
+You should name the file according to the experiment's Normandy slug,
+like bug_12345_my_cool_experiment.toml.
```
Comment (scholtzan) [inline README.md:24]:
```suggestion
like `bug_12345_my_cool_experiment.toml`.
```

File: README.md [hunk 2]
```diff
@@ -4,9 +4,152 @@ Custom configs for experiments analyzed in [jetstream](https://github.com/mozill
...
+[experiment]
+# Nominal length of the enrollment period in days.
+# Mozanalysis will consider enrollment_period + 1 "dates" of enrollments.
+enrollment_period = 7
+
+# The name of the control branch.
+# To compare all branches to each other pairwise, use the value `nan`.
+reference_branch = "control"
+
+# Override the "enrollment query" the mozanalysis will use.
```
Comment (scholtzan) [inline README.md:43]:
```suggestion
# Override the "enrollment query" mozanalysis will use.
```

File: README.md [hunk 3]
```diff
@@ -4,9 +4,152 @@ Custom configs for experiments analyzed in [jetstream](https://github.com/mozill
...
+      DATE(submission_timestamp) BETWEEN {{experiment.start_date}} AND {{experiment.enrollment_end}}
+      AND mozfun.map.get_key(environment.experiments, "{{experiment.slug}}") IS NOT NULL
+"""
+```
+
+### Metrics section
+
+The metrics section allows you to specify and define the metrics that you're collecting,
+the statistical summaries that you'd like applied to them, and any filters that you need.
+See DTMO for more details on the analysis window concept.
```
Comment (scholtzan) [inline README.md:65]: We could maybe add a link for this.

File: README.md [hunk 4]
```diff
@@ -4,9 +4,152 @@ Custom configs for experiments analyzed in [jetstream](https://github.com/mozill
...
+For example, you could define a new metric based on a scalar named
+`browser.engagement.cows_clicked` this way:
+
+```toml
+[metrics.ever_clicked_cows]
+# A clause of a SELECT expression. The expression must contain an aggregation function.
+# The expression is evaluated with `GROUP BY client_id, branch` over an analysis window.
+# Interpreted as a Jinja2 template. The mozanalysis helper functions are available,
+# so you could equivalently write the expression below like:
+# select_expression = "{{agg_sum("payload.processes.parent.scalars.browser_engagment_cows_clicked")} > 0"
```
Comment (scholtzan) [inline README.md:99]:
```suggestion
# select_expression = "{{agg_sum("payload.processes.parent.scalars.browser_engagment_cows_clicked")}} > 0"
```

File: README.md [hunk 5]
```diff
@@ -4,9 +4,152 @@ Custom configs for experiments analyzed in [jetstream](https://github.com/mozill
...
+
+# The data source to use. You can use the slug of a data source defined in mozanalysis,
+# or else define a new data source below.
+data_source = "main"
+```
+
+You should also add some sections to describe how your new metrics should be summarized for reporting.
+You can do this by adding a statistics section to the metric for the statistic you want.
+
+There is a list of statistics in the Jetstream wiki.
```
Comment (scholtzan) [inline README.md:111]: Is there? 😉

File: README.md (hunk 4 above)
Comment (tdsmith) [reply README.md]: good catch; counting is hard

File: README.md (hunk 5 above)
Comment (tdsmith) [reply README.md:111]: not yet 😅 I should at least create the page and link it here.

**Pull request merged**